lib/widgets/info_card.dart

```diff
@@ -18,85 +19,49 @@ class InfoCard<T> extends StatelessWidget {
   final String title;
   final ValueKey<String> showMoreButtonKey;
   final EdgeInsetsGeometry padding;
+  final bool showIfEmpty;
 
   @override
   Widget build(final BuildContext context) {
-    return Padding(
-      padding: padding ?? const EdgeInsets.fromLTRB(12, 12, 12, 0),
-      child: Card(
-        child: Padding(
-          padding: const EdgeInsets.all(12),
-          child: Column(
-            children: <Widget>[
-              Row(
-                mainAxisAlignment: MainAxisAlignment.spaceBetween,
-                children: [
-                  if (title != null && title.isNotEmpty)
-                    Expanded(
-                      child: Text(
-                        title,
-                        style: Theme.of(context)
-                            .textTheme
-                            .headline6
-                            .copyWith(fontSize: 18),
-                        overflow: TextOverflow.fade,
-                        maxLines: 2,
-                      ),
-                    ),
-                  if (onShowMore != null)
-                    GestureDetector(
-                      onTap: onShowMore,
-                      key: showMoreButtonKey ?? const ValueKey('show_more'),
-                      child: Row(
-                        children: <Widget>[
-                          Text(
-                            S.current.actionShowMore,
-                            style: Theme.of(context)
-                                .coloredTextTheme
-                                .subtitle2
-                                .copyWith(
-                                    color: Theme.of(context).primaryColor),
-                          ),
-                          Icon(
-                            Icons.arrow_forward_ios_outlined,
-                            color: Theme.of(context).primaryColor,
-                            size:
-                                Theme.of(context).textTheme.subtitle2.fontSize,
-                          )
-                        ],
-                      ),
-                    ),
-                ],
-              ),
-              const SizedBox(height: 10),
-              FutureBuilder(
-                  future: future,
-                  builder: (final context, final snapshot) {
-                    if (snapshot.connectionState == ConnectionState.done) {
-                      if (snapshot.hasData) {
-                        if ((snapshot.data is Map ||
-                                snapshot.data is Iterable) &&
-                            snapshot.data.isEmpty) {
-                          return noneYet(context);
-                        }
+    return FutureBuilder(
+        future: future,
+        builder: (final context, final snapshot) {
+          if (snapshot.connectionState == ConnectionState.done) {
+            // this condition resolves to true if the snapshot contains some data and if that data is iterable and not empty
```

Comment:
This comment basically echoes the code, I don't think it's necessary. A good variable name is preferable over a comment most times.